mutex.go

```diff
@@ -108,35 +106,48 @@ func (m *mutex) Unlock() {
 			if err == nil {
 				res.Body.Close()
 				if res.StatusCode == 204 {
-					return
+					// Check for timeout before returning.
+					okToReturn = true
```

Comment:
Same here. Not sure why is delaying successful return is needed.
I think it's supposed to return as quickly as possible on success.